File: BOR2/apps/api/internal/domain/qbtime.go
package domain

import "time"

// QBTimeTeam is a named group of employees for a given company.
type QBTimeTeam struct {
	ID        string    `json:"id"`
	Company   string    `json:"company"`
	Name      string    `json:"name"`
	Members   []string  `json:"members"`
	CreatedAt time.Time `json:"createdAt"`
}

// QBTimeEmployeeTeam links one QB Time employee to the team QB Time itself
// reports them under (synced daily via the QBT Groups API), with an optional
// manual override that takes precedence and is flagged as a divergence from
// what QuickBooks Time currently shows.
type QBTimeEmployeeTeam struct {
	ID                string     `json:"id"`
	Company           string     `json:"company"`
	QBTUserID         int        `json:"qbtUserId"`
	EmployeeName      string     `json:"employeeName"`
	QBTeamID          *int       `json:"qbTeamId"`
	QBTeamName        *string    `json:"qbTeamName"`
	OverrideTeamName  *string    `json:"overrideTeamName"`
	OverriddenBy      *string    `json:"overriddenBy"`
	OverriddenAt      *time.Time `json:"overriddenAt"`
	LastSyncedAt      time.Time  `json:"lastSyncedAt"`
	EffectiveTeamName string     `json:"effectiveTeamName"`
	IsOverridden      bool       `json:"isOverridden"`
}

File: BOR2/apps/web/src/types/shared/qbtime.ts
export interface QBTimeDailyReport {
  id: string
  company: string
  date: string        // "YYYY-MM-DD"
  fileName: string
  entries?: QBTimeDailyReportEntry[]
  createdAt: string
  createdById: string
  createdByName: string
}

export interface QBTimeDailyReportEntry {
  id: string
  reportId: string
  employeeRaw: string     // "Last, First"
  employeeDisplay: string // "First Last"
  jobCode: string
  regularHours: number
  overtimeHours: number
  totalHours: number
}

export interface QBTimeTeam {
  id: string
  company: string
  name: string
  members: string[]
  createdAt: string
}

export interface QBTimeEmployeeTeam {
  id: string
  company: string
  qbtUserId: number
  employeeName: string
  qbTeamId: number | null
  qbTeamName: string | null
  overrideTeamName: string | null
  overriddenBy: string | null
  overriddenAt: string | null
  lastSyncedAt: string
  effectiveTeamName: string
  isOverridden: boolean
}
